drivers/peci/peci_handlers.c

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/peci.h>
+#include <syscall_handler.h>
+
+static inline int z_vrfy_peci_config(struct device *dev, u32_t bitrate)
+{
+	Z_OOPS(Z_SYSCALL_DRIVER_PECI(dev, config));
+
+	return z_impl_peci_config((struct device *)dev, (u16_t)bitrate);
+}
+#include <syscalls/peci_config_mrsh.c>
+
+static inline int z_vrfy_peci_enable(struct device *dev)
+{
+	Z_OOPS(Z_SYSCALL_DRIVER_PECI(dev, enable));
+
+	return z_impl_peci_enable((struct device *)dev);
+}
+#include <syscalls/peci_enable_mrsh.c>
+
+static inline int z_vrfy_peci_disable(struct device *dev)
+{
+	Z_OOPS(Z_SYSCALL_DRIVER_PECI(dev, disable));
+
+	return z_impl_peci_disable((struct device *)dev);
+}
+#include <syscalls/peci_disable_mrsh.c>
+
+static inline int z_vrfy_peci_transfer(struct device *dev,
+				       const struct peci_msg *msg)
+{
+	struct peci_msg msg_copy;
+
+	Z_OOPS(Z_SYSCALL_DRIVER_PECI(dev, transfer));
+
+	Z_OOPS(Z_SYSCALL_MEMORY_READ((const struct peci_msg *)msg,
+				      sizeof(struct peci_msg)));
+
+	Z_OOPS(Z_SYSCALL_MEMORY_READ(((struct peci_msg *)msg)->tx_buffer->buf,
+				     ((struct peci_msg *)msg)->tx_buffer->len));
+
+	Z_OOPS(z_user_from_copy(&msg_copy, msg, sizeof(msg_copy)));
+
+	Z_OOPS(Z_SYSCALL_MEMORY_WRITE(((struct peci_msg *)msg)->rx_buffer->buf,
+				     ((struct peci_msg *)msg)->rx_buffer->len));
+
+	return z_impl_peci_transfer((struct device *)dev,
+				    (const struct peci_msg *)&msg_copy);
```

Comment:
Why do you have so many casts to (const struct peci_msg *) that shouldn't be necessary.

Reply:
Just following  the counter_handlers.c, adc_handlers.c I though it was a style requirement. I can remove them